### PR TITLE
Utility to download OSV.dev source records

### PR DESCRIPTION
**Pull request by andrewpollock**
To aid with running OSV record lint checks against all of a source's records, pre-import

### CODE REVIEW

File: tools/source-sync/source_sync.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+from urllib3.util.retry import Retry
+import requests
+from requests.adapters import HTTPAdapter
+import re
+import pygit2
+from google.cloud import storage
+from google.cloud.storage import retry
+
+
+class Error(Exception):
```
Comment (oliverchang) [inline tools/source-sync/source_sync.py:19]: nit: use a docstring here (and then we can also get rid of the `pass`)

File: tools/source-sync/source_sync.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  pass
+
+
+def Sources(source: str) -> dict:
```
Comment (oliverchang) [inline tools/source-sync/source_sync.py:23]: nit: s/source/source_path/ to be more explicit about what this argument means?

**Review by oliverchang — APPROVED** [review]

File: tools/source-sync/source_sync.py (hunk 1 above)
Comment (andrewpollock) [reply tools/source-sync/source_sync.py:19]: Done in https://github.com/google/osv.dev/pull/2416/commits/159714a2edc34dfdd1d0b92fb6f9cdddfce632ee

File: tools/source-sync/source_sync.py (hunk 2 above)
Comment (andrewpollock) [reply tools/source-sync/source_sync.py]: Done in https://github.com/google/osv.dev/pull/2416/commits/159714a2edc34dfdd1d0b92fb6f9cdddfce632ee

**Pull request merged**